### PR TITLE
Allow ClusterOperator to be set to Degraded state

### PR DESCRIPTION
**Pull request by sadasu**
Set CO to Degraded state when provisioning CR validations fail

### CODE REVIEW

File: controllers/clusteroperator.go [hunk 1]
```diff
@@ -149,3 +149,24 @@ func (r *ProvisioningReconciler) updateCOStatusDisabled() error {
...
+
+// updateCOStatusDegraded updates the ClusterOperator's Degraded
+// degradedReason should contain the current reason for the Operator to be marked in that state
+func (r *ProvisioningReconciler) updateCOStatusDegraded(degradedReason string, detailedError string) error {
+	degradedMessage := "Operator is Degraded"
+	progressingMessage := "Operator is Degraded while Progressing"
+
+	co, err := r.getOrCreateClusterOperator()
+	if err != nil {
+		r.Log.Error(err, "failed to get or create ClusterOperator")
```
Comment (dhellmann) [inline controllers/clusteroperator.go:161]: Is logging here before returning the error going to mean the error is logged multiple times? Should this just wrap the error with more detail and then return it?

**Review by dhellmann — COMMENTED** [review]
This looks good. I have one question about error handling inline.

File: controllers/clusteroperator.go (hunk 1 above)
Comment (sadasu) [reply controllers/clusteroperator.go:161]: Removed this error msg. getOrCreateClusterOperator() is returning a good error message that can be wrapped with one  from Reconcile() and returned from there. Doing the same for updateCOStatusDisabled().

**Pull request merged**